### PR TITLE
feat: cli demo

### PR DESCRIPTION
**Pull request by flekschas**
Inspired by @manzt great work on getting a quick start demo up for [cev](https://github.com/OzetteTech/comparative-embedding-visualization), this PR adds a CLI for starting a demo notebook. Apart from that this PR fixes a regression from https://github.com/flekschas/jupyter-scatter/pull/170

## Description

> What was changed in this pull request?

- Added a CLI script: `uv run jupyter-scatter`. This can be used to quickly get started with Jupyter via `uvx jupyter-scatter demo`

> Why is it necessary?

Because it's cool!

## Checklist

- [x] Provided a concise title as a [semantic commit message](https://www.conventionalcommits.org) (e.g. "fix: correctly handle undefined properties")
- [x] `CHANGELOG.md` updated
- [ ] Tests added or updated
- [x] Documentation in `API.md`/`README.md` added or updated
- [ ] Example(s) added or updated
- [ ] Screenshot, gif, or video attached for visual changes

### CODE REVIEW

File: jscatter/_cli.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        sys.exit(1)
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="jupyter-scatter")
+    subparsers = parser.add_subparsers(dest="command", help="Available commands")
+    subparsers.add_parser("demo", help=f"Run the demo notebook in JupyterLab")
+    args = parser.parse_args()
+
+    notebook_path = Path("notebooks/demo.ipynb")
```
Comment (manzt) [inline jscatter/_cli.py:54]: I think you'll want to make this path resolved relative to this file:

```suggestion
    notebook_path = Path(__file__) "../notebooks/demo.ipynb"
```

and also make sure that the notebook is included in the hatchling wheel and sdist.


My guess is that right now if you run:

```sh
uv build
uvx --from=./dist/jupyter-scatter-<wheel> jupyter-scatter demo
```

where will be an error.

File: jscatter/_cli.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        "jupyterlab",
+        "--with",
+        "." if _DEV else f"jupyter-scatter=={__version__}",
+        "jupyter",
+        "lab",
+        str(notebook_path),
+    ]
+
+    try:
+        os.execvp(command[0], command)
```
Comment (manzt) [inline jscatter/_cli.py:42]: maybe for a follow-up PR, but process substitution only works on unix systems. This will fail on windows.

Comment (flekschas) [reply jscatter/_cli.py:42]: Does anybody still use windows? 🤔🤭🤫

Thanks for bringing this up! Is this the same issue that came up in `juv`?

File: jscatter/_cli.py (hunk 1 above)
Comment (flekschas) [reply jscatter/_cli.py]: Thanks for the catch!

File: jscatter/_cli.py (hunk 2 above)
Comment (manzt) [reply jscatter/_cli.py:42]: > Does anybody still use windows? 🤔🤭🤫

Lots!

> Thanks for bringing this up! Is this the same issue that came up in juv?

Yeah, it also came up in juv. It's a non-trivial problem, especially with Jupyter. The solutions I've found require a `subprocess`, but the issue is that you also need to intercept keyboard interrupts to gracefully exit. Also I'm not sure the executable is called `uv` on windows (e.g., `uv.exe`).  

Here's an example from `uv`:

- https://github.com/astral-sh/uv/blob/main/python/uv/__main__.py

They delegate whether to use `subprocess` or `execvpe` based on `sys.platform`. In `juv` we ended up just using `subprocess` in either case:

- https://github.com/manzt/juv/blob/main/src/juv/_run_replace.py

Comment (flekschas) [reply jscatter/_cli.py]: Thanks for sharing! I ended up implementing a strategy similar to `uv`'s. Hopefully it's working for windows folks.

File: jscatter/_cli.py (hunk 1 above)
Comment (flekschas) [reply jscatter/_cli.py]: Turns out that including the file in the wheel is more complicated than expected. I'm not a Python packaging expert but even if a folder outside the actual package (`jscatter`) is included in the wheel, it ends up not being accessible after the wheel is installed. I ended up dynamically downloading the notebook from GitHub to get around the packaging issue.

**Pull request merged**